### DISCUSSION TITLE
Quantization

### DISCUSSION THREAD

**Discussion by d3ztr0yur3000** [tracel-ai/burn#1079] [Ideas]
As this library is weighted toward performance and resource efficiency are there any plans to support Quantization and model formats such as GGUF? Will you also support quantized training as well using methods such as QLoRA?

**Comment by nathanielsimard** [comment]
Yes Quantization is planned for next year! We also want to support QLora as well, but I'm not sure when it will land.

**Comment by hrstoyanov** [comment]
Burn is awesome!
Is there any progress on quantization ?

**Reply by laggui** [reply]
We just added better quantization support with persistent memory optimization in the last release.

Right now, model weights can be quantized and operations are performed with float precision signal.

Most operations are not defined yet for quantized tensors, so the values have to be dequantized. But with fusion this can be fused with pretty much any other op, so we have reduced global memory reads and writes.

Matmul is the only compute operation currently defined with explicit support for quantized inputs. More to come in the future!

More info here: https://burn.dev/blog/release-0.19.0/#quantization

And some practical numbers in our reddit post: https://www.reddit.com/r/rust/comments/1oiexhr/burn_0190_release_quantization_distributed/
